File: Python/String-2/count_hi.py


'''

Return the number of times that the string "hi" appears anywhere in the given string.


count_hi('abc hi ho') -> 1
count_hi('ABChi hi') -> 2
count_hi('hihi') -> 2

'''

def count_hi(str):
  
    count = 0
  
    for i in range(len(str)-1):
        z = i+1
        
        if(str[i]=='h' and str[i+1]=='i'):
            count +=1

    return count 

class Test_count_hi(unittest.TestCase):

    def test_1(self):
        self.assertEqual(count_code('aaacodebbb'), 1)

    def test_2(self):
        self.assertEqual(count_code('codexxcode'), 2)

    def test_3(self):
        self.assertEqual(count_code('cozexxcope'), 0)


if __name__ == '__main__':
    unittest.main() 

File: Java/String_2/countHi.java
/**
 * @author Luis Teixeira
 *
 * @date   02/04/2018
 *
 * @mail   filipe.teixeira.996@gmail.com
 */
package String_2;

import static org.junit.Assert.*;

import org.junit.Test;

/**

Return the number of times that the string "hi" appears anywhere in the given string.


countHi("abc hi ho") --> 1
countHi("ABChi hi") --> 2
countHi("hihi") --> 2 
 */
public class countHi {

	private static int i = 1;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countHi(\"abc hi ho\") --> 1\n");
		assertEquals("countHi(\"abc hi ho\") --> 1", 1, countHi("abc hi ho"));
		i++;
	}
	
	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countHi(\"ABChi hi\") --> 2\n");
		assertEquals("countHi(\"ABChi hi\") --> 2", 2, countHi("ABChi hi"));
		i++;
	}
	
	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countHi(\"hihi\") --> 2 \n");
		assertEquals("countHi(\"hihi\") --> 2 ", 2, countHi("hihi"));
		i++;
	}
	
	public int countHi(String str) {
	
		int val = 0;
		
		for (int i = 0; i < str.length(); i++) {
			
		}
	}
}
